zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/BackupManager.java

```diff
@@ -63,9 +64,12 @@
   private final BackupStorageProvider backupStorage;
   private final long serverId;
   private final String namespace;
-  private BackupBean backupBean = null;
+  @VisibleForTesting
+  protected BackupBean backupBean = null;
   private BackupStats backupStats = null;
 
+  private int snapshotFilesBackedUp;
```

Comment:
Nit: `numSnapshotFilesBackedUp`?

Reply:
This is inside`backupStats`, it's the `getNumberOfSnapshotFilesBackedUpLastIteration` metric. The field here is for calculation purpose, so within one iteration we can increment this number every time one more snapshot file is backed up and we can report this number to the `getNumberOfSnapshotFilesBackedUpLastIteration` metric and reset this number when an iteration ends.

Reply:
What I meant is that the scope of this internal variable lies within backupStats (conceptually) - BackupManager does not need to know this so it doesn't need access to it. You could add helper functions like `incrementNumSnapshotFilesBackedUp()` to update `BackupStats`'s internal data.

Reply:
Good point! Updated the code.